src/org/mozilla/javascript/Parser.java

```diff
@@ -1913,11 +1925,19 @@ private AstNode returnOrYield(int tt, boolean exprContext)
         consumeToken();
         int lineno = ts.lineno, pos = ts.tokenBeg, end = ts.tokenEnd;
 
+        boolean yieldStar = false;
+        if ((tt == Token.YIELD) &&
+            (compilerEnv.getLanguageVersion() >= Context.VERSION_ES6) &&
+            (peekToken() == Token.MUL)) {
+          yieldStar = true;
+          consumeToken();
+        }
+
         AstNode e = null;
         // This is ugly, but we don't want to require a semicolon.
         switch (peekTokenOrEOL()) {
           case Token.SEMI: case Token.RC:  case Token.RB:    case Token.RP:
-          case Token.EOF:  case Token.EOL: case Token.ERROR: case Token.YIELD:
```

Comment:
Is this Lang version dependant?

Reply:
I don't think so because "yield" has been a supported keyword for many Rhino releases now. Let me run a quick test and check.